pkg/operator/controller.go

```diff
@@ -676,6 +677,11 @@ func (c *Controller) syncProfile(tuned *tunedv1.Tuned, nodeName string) error {
 		return fmt.Errorf("failed to get ProviderName: %v", err)
 	}
 
+	// Won't be set in case of operator restart
+	if _, ok := c.pc.state.profileLastUpdatedGeneration[profile.Name]; !ok {
+		c.pc.state.profileLastUpdatedGeneration[profile.Name] = profile.Generation + 1
```

Comment:
I didn't test, but are you sure we want to initialize to `profile.Generation + 1`, not just `profile.Generation` ?

Reply:
I think you're right, good catch. Still testing this...